### ISSUE TITLE
[Feature] Bypass Press-and-Hold HUMAN challange

### ISSUE THREAD

**Issue by gabrielemidulla** [berstend/puppeteer-extra#893]
**Feature request**

I'm trying to create an e-commerce WebScraper, but the website I'm targeting uses a HUMAN Press-and-Hold captcha. I have tried simulating user actions to bypass it, but it seems that there is some underlying mechanism to detect bots. 

I have looked into the documentation of humansecurity (https://edocs.humansecurity.com/docs/human-challenge#test-scenarios), but I couldn't find a solution. 

It would be great if there was a way to bypass these non-mainstream captchas as well. 

HUMAN security solutions are widely used, so having a bypasser for HUMAN security captchas would be a powerful addition to the plugin.

E-commerce: https://wayfair.com

![bb54d11-press_and_hold](https://github.com/berstend/puppeteer-extra/assets/56894431/50874cb8-f1a6-40e0-9412-f84d7178761a)

**Comment by vladtreny** [comment]
We have solved it recently, but in the private community. 
In short, need to rewrite a lot... or you need an anti-detect that solved it.  
This lib will not help you.

**Comment by gabrielemidulla** [comment]
What do you mean for private community? Where can I download the bypasser?

**Comment by vladtreny** [comment]
I tried to reproduce. Is it correct?

It is my internal bypasser. 

https://github.com/berstend/puppeteer-extra/assets/12267987/3710a540-5917-4593-bd23-59194e916d78

**Comment by gabrielemidulla** [comment]
Yeah that works as well, but the captcha shows all the time! I don't even know which data is being used by the HUMAN challenge to detect whether im a bot or not...

**Comment by vladtreny** [comment]
They detect your cdp + bad IP

**Comment by gabrielemidulla** [comment]
is CDP Chrome DevTools Protocol?

**Comment by vladtreny** [comment]
I do not get this captcha. To reproduce the captcha, I had to disable cdp evasions and visit from a very bad proxy.

**Comment by gabrielemidulla** [comment]
Sorry, what does CDP stands for? Either way, I also thought that rotating proxies could do the trick

**Comment by vladtreny** [comment]
If you clicked before, they seem like bypass automatically on this IP. 

There is another variant of this captcha. Do not know which is newer

https://github.com/berstend/puppeteer-extra/assets/12267987/4aba8026-ff1e-47cf-bff1-d427484482ed

**Comment by gabrielemidulla** [comment]
No problem, what does UA mean?

**Comment by vladtreny** [comment]
In summary, you need to solve cdp detection.

**Comment by gabrielemidulla** [comment]
> In summary, you need to solve cdp detection. 

Alright, could you tell me what does CDP stand for?

**Comment by Maxim-Mazurok** [comment]
> is CDP Chrome DevTools Protocol?

Correct, see https://github.com/berstend/puppeteer-extra/issues/899

**Comment by vladtreny** [comment]
It is not possible to fix it properly without modifying the Chromium source.

If anyone could start a project for a custom Chromium, would be great

**Comment by GiveDaData** [comment]
> It is not possible to fix it properly without modifying the Chromium source.
> 
> If anyone could start a project for a custom Chromium, would be great

I have solved the CDP detection without modifying any source code, but since I use it for my business, it does not make sense to share it publicly, just like you are not sharing your solution

**Comment by vladtreny** [comment]
@GiveDaData

I agree publishing makes no sense because these threads are monitored. 

But anyway, cdp solution is not ideal. If you try to solve this ^ captcha, you will see issues. I bypass this captcha (by click), but yes, I saw limitations.  

Ideal solution will be patching chromium and using puppeteer without any evasions at all.

Btw, if you want to reproduce the captcha, you can do following
```
await page.setUserAgent('HeadlessChrome')
await page.goto('https://www.wayfair.com/v/account/authentication/create')

```

**Comment by GiveDaData** [comment]
> @GiveDaData
> 
> I agree publishing makes no sense because these threads are monitored.
> 
> But anyway, cdp solution is not ideal. If you try to solve this ^ captcha, you will see issues. I bypass this captcha (by click), but yes, I saw limitations.
> 
> Ideal solution will be patching chromium and using puppeteer without any evasions at all.
> 
> Btw, if you want to reproduce the captcha, you can do following
> 
> ```
> await page.setUserAgent('HeadlessChrome')
> await page.goto('https://www.wayfair.com/v/account/authentication/create')
> ```

I just tried this link 'https://www.wayfair.com/v/account/authentication/create' with my bot, and it can go through without hitting any capthca or Press-andHold challange. my email was receiving the verification code

**Comment by vladtreny** [comment]
I also bypass it. But I wanted to solve this captcha. 

So had to set `await page.setUserAgent('HeadlessChrome')` to get the captcha intentionally. Then you can remove it, and you will see captcha on every load. 

I solved it.. but yes, good job Human, if you read this   😅
Someone very smart made it.

**Comment by marcpre** [comment]
> We have solved it recently, but in the private community. In short, need to rewrite a lot... or you need an anti-detect that solved it. This lib will not help you.

Which private community are you talking about? Would love to be part of it.

**Comment by klimsava** [comment]
Has someone found a solution to solve this captcha?

**Comment by nomaam** [comment]
I am also looking for a solution to this challenge

**Comment by bhdupati-fadv** [comment]
Any solution for this yet?

**Comment by vladtreny** [comment]
You will not pass it with this lib. They specifically detect it.

**Comment by bhdupati-fadv** [comment]
ok @vladtreny .then what is the solution. can you please share any solution if already solved it

**Comment by klimsava** [comment]
> You will not pass it with this lib. They specifically detect it.

Do you know an alternative solution?

**Comment by bhdupati-fadv** [comment]
@klimsava i have used playwrightextrasharp2 , this uses stealth plugin with playwright. but this also didnt solve.
Did you try any alternate solutions?
